_coffee/PeopleSearch.coffee

```diff
@@ -0,0 +1,132 @@
+class PeopleSearch
+  SEARCH_INDEX_URL = '/feeds/people_index.json'
+  PEOPLE_INDEX_URL = '/feeds/people2.json'
+  FIELDS = [
+    ['people-filter-name', '']
+    ['people-filter-graduated', 'graduated']
+    ['people-filter-career', 'career']
+    ['people-filter-course', 'course']
+    ['people-filter-award', 'award']
+    ['people-filter-srole', 'srole']
+    ['people-filter-crole', 'crole']
+  ]
+
+  PS_FILTER_FIXED_TOP = 27
+  PS_FILTER_FIXED_CLASS = 'people-index__filters--fixed'
+
+  constructor: (opts) ->
+    @enabled = true
+
+    @psFilterEl = opts.psFilterEl
+    @psResults = new PeopleResults
+      psResultsEl: opts.psResultsEl
+
+    @footerEl = document.querySelector('.site-footer')
+
+    @searchIndex = new Object
+    @peopleIndex = new Object
+    @peopleReverseIndex = new Object
+    @loadIndex()
+
+    @bindSearchFields()
+
+    @psFilterElOffsetTop = $(@psFilterEl).offset().top
+    window.requestAnimationFrame(@update)
+
+  destructor: ->
+    @enabled = false
+
+  loadIndex: ->
```

Comment:
Should abstract the searchy bits into parent class. Will enable site search to use new class based search.